### DISCUSSION TITLE
how to encrypt using aws kms in sqlalchemy

### DISCUSSION THREAD

**Discussion by KIWONY** [sqlalchemy/sqlalchemy#10885] [Usage Questions]
I am trying to implement encryption using SQLAlchemy when storing data in PostgreSQL. I referred to the code for StringEncryptedType in the documentation of sqlalchemy-utils, which I have attached via the link.

[https://sqlalchemy-utils.readthedocs.io/en/latest/data_types.html#sqlalchemy_utils.types.encrypted.encrypted_type.StringEncryptedType](url)

```python
class Test(Base):
    __tablename__ = "test"

    id = Column(Integer, primary_key=True)
    name = Column(StringEncryptedType(String, key_id, AesEngine, 'pkcs5'))
```

But I would like to use AWS KMS here and when I set the ARN of AWS KMS as the secret key, the data is not encrypted.
What am I missing?

**Comment by zzzeek** [comment]
hi- 

any chance you can reach out on the [sqlalchemy-utils](https://github.com/kvesteri/sqlalchemy-utils/issues) tracker?  we dont maintain that project here

**Reply by KIWONY** [reply]
ahah sorry, thank you!!!
